### PR TITLE
ARROW-3932: [Python] Include Benchmarks.md in Sphinx docs

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: docs/source/python/index.rst [hunk 1]
```diff
@@ -47,3 +47,4 @@ files into Arrow structures.
    api
    development
    getting_involved
+   benchmarks
```
Comment (pitrou) [inline docs/source/python/index.rst]: Perhaps we may want to group "development", "getting involved" and "benchmarks" in a subchapter.

File: python/README-benchmarks.md [hunk 2]
```diff
@@ -1,47 +0,0 @@
...
-
-  Unless required by applicable law or agreed to in writing,
-  software distributed under the License is distributed on an
-  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-  KIND, either express or implied.  See the License for the
-  specific language governing permissions and limitations
-  under the License.
--->
-
-# Benchmarks
```
Comment (pitrou) [inline python/README-benchmarks.md]: Perhaps keep the README file but just put a redirect to the Sphinx docs?
(which, unfortunately, aren't online yet...) @wesm

Comment (xhochy) [reply python/README-benchmarks.md]: Manually pushed them: http://arrow.apache.org/docs/latest/

They are not linked from anywhere yet though.

File: docs/source/python/index.rst (hunk 1 above)
Comment (xhochy) [reply docs/source/python/index.rst]: Is there a simple way to do this in a `toctree` or should we start a subfolder `contributing`?

**Comment by xhochy** [comment]
+1, merging. I can address additional comments in a follow-up PR.

**Pull request closed without merging**